module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -726,13 +726,13 @@ func testAddingEntities(t *testing.T, pool *Pool, entitiesToBeAdded []*unittest.
 			usedTraverseStep = uint32(len(pool.poolEntities))
 		}
 		tailAccessibleFromHead(t,
-			pool.used.head,
-			pool.used.tail,
+			pool.states[stateUsed].head,
```

Comment:
nit: Should add funcs to `states` like used() and free() since we use pool.states[SOME_INDEX] often.

Reply:
Just to clarify that I understood you correctly. You suggest that pool.states[stateUsed].head is converted to pool.states.used().head ?
or pool.used().head - if a used() method is implemented in the pool since the pool has control over a number of states and state for now has fields allowing it to represent any linked list with in the pool.